tests/integ/test_dataframe.py

```diff
@@ -1135,29 +1135,48 @@ def test_create_dataframe_with_basic_data_types(session):
 
 def test_create_dataframe_with_semi_structured_data_types(session):
     data = [
-        ["'", 2],
-        ("'", 2),
-        [[1, 2], [2, 1]],
-        array("I", [1, 2, 3]),
-        {"'": 1},
+        [
+            ["'", 2],
+            ("'", 2),
+            [[1, 2], [2, 1]],
+            array("I", [1, 2, 3]),
+            {"'": 1},
+        ],
+        [
```

Comment:
why do we need another row here?

Reply:
This is mainly to make sure merge type logic is covered for complex type: https://github.com/snowflakedb/snowpark-python/blob/8c52606a44096a265ddbdcd4d514e1ca8d51b9a1/src/snowflake/snowpark/_internal/type_utils.py#L320

The coverage report said that the `MapType` is not covered, and it feels like this is the easiest fix :)